app/models/user.rb

```diff
@@ -116,6 +116,11 @@ def blog_slug
   enum role: {
     member: 0,
     admin: 1,
-    applicant: 2
+    applicant: 2,
+    moderator: 3
   }
+
+  def application_reviewer?
```

Comment:
This should live inside the policy, not in the model.

Reply:
The only problem with putting it in the `user_only` policy is that it's also needed [here](https://github.com/agency-of-learning/PairApp/pull/329/files#diff-a5be11d25ddf72ce693ca15c7de9bd8760a1923931bafdfae5e9be70d27f0978R7) inside of the `UserMenteeApplicationPolicy`.

I'd rather avoid duplicating the method definition across multiple policies. Another solution would be to delegate the call in `UserMenteeApplicationPolicy` to `UserOnlyPolicy` ie:
```rb
def show?
  application_reviewer? || matching_user?
end

private

def application_reviewer?
  # imo this is a bit verbose for what's being expressed
  # wouldn't hate cleaning it up with some inheritance or mixin strategy if this will be a frequent pattern
  Pundit.policy(user, :user_only).application_reviewer?
end
```

Reply:
I'd move this to the ApplicationPolicy base class honestly for now. We can figure out how to clean this up more later on.